include/BothAnalogDigital.h

```diff
@@ -0,0 +1,30 @@
+#ifndef BOTHANALOGDIGITAL_H__
+#define BOTHANALOGDIGITAL_H__
+
+#include "mbed.h"
+
+
+// storing every digital data inside a struct to reduce the memory
+struct Digital_Data{
+    bool setCruiseControlReading:1;
+    bool resetCruiseControlReading:1;
+    bool cruisePowerModeReading:1;
+    bool cruiseSpeedModeReading:1;
+    bool speedReading:1;
+    bool motorPowerReading:1;
+    bool forwardAndReverseReading:1;
+    bool ControlforwardAndReverseReading:1;
+    bool ecoModeReading:1;
+};
+
+// digital struct to access the digital data
+extern volatile struct Digital_Data digital_data;
+
+// analog data
+extern volatile float acceleratorPedalReading;
+extern volatile float acceleratorSignalReading;
+extern volatile float brakeStatusReading;
+extern volatile float regenerativeBrakingReading;
```

Comment:
Nit: Remove the word "Reading" from the end, it doesn't give any more detail to what the signal is